### PR TITLE
feat(share_plus): remove direct dependence of url_launcher

### LINKED ISSUE #1070: [Request]: share_plus android remove url_launcher
### Plugin

share_plus

### Use case

url_launcher is not required when compiling android but it will also be loaded into the app's gradle project

### Proposal

Remove url_launcher when compiling android

### PR DESCRIPTION
**Pull request by SirusCodes**
## Description

Remove direct dependence of url_launcher, instead depending on platform specific packages.

## Related Issues

resolves #1070

## Checklist

- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [ ] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by miquelbeltran** [comment]
Is this PR still in progress?

**Comment by SirusCodes** [comment]
Yes, I just need to fix some test.

**Review by miquelbeltran — APPROVED** [review]
I like the idea. I don't fully understand why it was a problem to have url_launcher as a dependency originally, but I like that now it is injected as param, easier to mock or replace even.

**Comment by miquelbeltran** [comment]
I will have to test on Monday before merging

**Comment by SirusCodes** [comment]
> I don't fully understand why it was a problem to have url_launcher as a dependency originally

Someone wanted to save a few KBs in their app, so why not.

**Comment by miquelbeltran** [comment]
Thanks!

File: packages/share_plus/share_plus/lib/src/share_plus_linux.dart [hunk 1]
```diff
@@ -35,7 +40,11 @@ class SharePlusLinuxPlugin extends SharePlatform {
           .join('&'),
     );
 
-    await launchUrl(uri);
+    if (await urlLauncher.canLaunch(uri.toString())) {
+      await urlLauncher.launchUrl(uri.toString(), const LaunchOptions());
+    } else {
+      throw Exception('Unable to share on web');
+    }
   }
```
Comment (stuartmorgan) [inline packages/share_plus/share_plus/lib/src/share_plus_linux.dart:48]: FYI: changing to this construction will break functionality, and has no advantages; see https://pub.dev/packages/url_launcher#checking-supported-schemes

Comment (SirusCodes) [reply packages/share_plus/share_plus/lib/src/share_plus_linux.dart:48]: So, it's better to have only `launchUrl` in try catch block?

Comment (SirusCodes) [reply packages/share_plus/share_plus/lib/src/share_plus_linux.dart:48]: [Checking the return of launchUrl](https://pub.dev/packages/url_launcher#checking-supported-schemes:~:text=if%20(!await%20launchUrl(uri))%20%7B%0A%20%20throw%20%27Could%20not%20launch%20%24uri%27%3B%0A%7D) is better construct?

Comment (stuartmorgan) [reply packages/share_plus/share_plus/lib/src/share_plus_linux.dart:48]: > So, it's better to have only `launchUrl` in try catch block?

I'm not sure what the goal of a try/catch would be given that you are currently throwing.

> [Checking the return of launchUrl](https://pub.dev/packages/url_launcher#checking-supported-schemes:~:text=if%20(!await%20launchUrl(uri))%20%7B%0A%20%20throw%20%27Could%20not%20launch%20%24uri%27%3B%0A%7D) is better construct?

If you don't have a specific need to know in advance of trying (which is rare, and certainly isn't the case given the code here), checking what actually happened is better than using a potentially inaccurate prediction of what might happen, yes.

Comment (nohli) [reply packages/share_plus/share_plus/lib/src/share_plus_linux.dart:48]: @miquelbeltran FYI

Comment (miquelbeltran) [reply packages/share_plus/share_plus/lib/src/share_plus_linux.dart:48]: If anyone can send a PR with the fix, I can review and merge asap

Comment (SirusCodes) [reply packages/share_plus/share_plus/lib/src/share_plus_linux.dart:48]: Yeah I can do that.

**Pull request merged**